.github/workflows/release.yml

```diff
@@ -0,0 +1,90 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Create Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      snapshot:
+        description: 'is snapshot version?'
+        type: boolean
+        default: true
+        required: true
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+
+    strategy:
+      matrix:
+        os: ['ubuntu-latest']
+        go-version: [1.18.x]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+  
+      -
+        name: Set up Go 1.x
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go-version }}
+        id: go
+     
+      -
+        name: Compute release version
+        uses: AirVantage/github-actions/version-action@v37
+        id: get-version
+        with:
+          is-snapshot: ${{ github.event.inputs.snapshot == 'true' && '1' || '0' }} 
+        
+      -
+        name: GitHub Tag & Release
+        uses: AirVantage/github-actions/github-release-action@v37
+        id: create-release
+        with:
+          version: ${{ steps.get-version.outputs.version }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          draft: ${{ github.event.inputs.snapshot == 'true' && '1' || '0' }} 
+
+
+      -
+        name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+        with:
+          # Fetch whole repository to get release tags
+          # See https://github.com/actions/checkout/issues/701
+          fetch-depth: 0
+
+      # Option B: split checkout + local tags ?
+      #-
+      #  name: Tag local copy
+      #  run: git tag ${{ steps.get-version.outputs.version }}
+        
+      -
+        name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v3
```

Comment:
So does this step publish the artefacts somewhere? 
I've just had a quick look at GoReleaser but it's unclear to me.

Reply:
Yes,

you can see an example here: https://github.com/AirVantage/test-bplessis-workflow-go/releases/

goreleaser use the .goreleaser.yml file to decide what to build, it also use git tags to find the version (the tag must be present on the checked-out repo so i had some fun tweaking the process) and publishing the artifact correctly.